metadata-builders/src/main/java/com/linkedin/metadata/builders/search/DataProcessIndexBuilder.java

```diff
@@ -17,13 +24,65 @@ public DataProcessIndexBuilder() {
         super(Collections.singletonList(DataProcessSnapshot.class), DataProcessDocument.class);
     }
 
+    @Nonnull
+    private static String buildBrowsePath(@Nonnull DataProcessUrn urn) {
+        return ("/" + urn.getOriginEntity() + "/"  + urn.getOrchestrator() + "/" + urn.getNameEntity())
+            .replace('.', '/').toLowerCase();
+    }
+
+    @Nonnull
+    private static DataProcessDocument setUrnDerivedFields(@Nonnull DataProcessUrn urn) {
+        return new DataProcessDocument()
+            .setName(urn.getNameEntity())
+            .setOrchestrator(urn.getOrchestrator())
+            .setUrn(urn)
+            .setBrowsePaths(new StringArray(Collections.singletonList(buildBrowsePath(urn))));
+    }
+
+    @Nonnull
+    private DataProcessDocument getDocumentToUpdateFromAspect(@Nonnull DataProcessUrn urn, @Nonnull Ownership ownership) {
+        final StringArray owners = BuilderUtils.getCorpUserOwners(ownership);
+        return setUrnDerivedFields(urn)
+            .setHasOwners(!owners.isEmpty())
+            .setOwners(owners);
+    }
+
+    @Nonnull
+    private DataProcessDocument getDocumentToUpdateFromAspect(@Nonnull DataProcessUrn urn,
+        @Nonnull DataProcessInfo dataProcessInfo) {
+        return  setUrnDerivedFields(urn)
+            .setNumInputDatasets(dataProcessInfo.getInputs().size())
```

Comment:
Because `inputs` & `outputs` fields of `DataProcessInfo` model are optional, it would be good to first check if they are not null like below:
```
if (dataProcessInfo.hasInputs()) {
...
}
```

Reply:
Fixed.